<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Описание-проекта" data-toc-modified-id="Описание-проекта-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Описание проекта</a></span><ul class="toc-item"><li><span><a href="#Описание-данных" data-toc-modified-id="Описание-данных-1.1"><span class="toc-item-num">1.1&nbsp;&nbsp;</span>Описание данных</a></span></li><li><span><a href="#Цель-исследования:" data-toc-modified-id="Цель-исследования:-1.2"><span class="toc-item-num">1.2&nbsp;&nbsp;</span>Цель исследования:</a></span></li><li><span><a href="#Ход-исследования:" data-toc-modified-id="Ход-исследования:-1.3"><span class="toc-item-num">1.3&nbsp;&nbsp;</span>Ход исследования:</a></span></li></ul></li><li><span><a href="#Общая-информация-о-данных" data-toc-modified-id="Общая-информация-о-данных-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Общая информация о данных</a></span></li><li><span><a href="#Предобработка-данных" data-toc-modified-id="Предобработка-данных-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Предобработка данных</a></span><ul class="toc-item"><li><span><a href="#Работа-с-пропущенными-значениями" data-toc-modified-id="Работа-с-пропущенными-значениями-3.1"><span class="toc-item-num">3.1&nbsp;&nbsp;</span>Работа с пропущенными значениями</a></span></li><li><span><a href="#Обработка-аномальных-значений" data-toc-modified-id="Обработка-аномальных-значений-3.2"><span class="toc-item-num">3.2&nbsp;&nbsp;</span>Обработка аномальных значений</a></span></li><li><span><a href="#Изменение-типов-данных" data-toc-modified-id="Изменение-типов-данных-3.3"><span class="toc-item-num">3.3&nbsp;&nbsp;</span>Изменение типов данных</a></span></li><li><span><a href="#Обработка-дубликатов" data-toc-modified-id="Обработка-дубликатов-3.4"><span class="toc-item-num">3.4&nbsp;&nbsp;</span>Обработка дубликатов</a></span></li><li><span><a href="#Категоризация-данных" data-toc-modified-id="Категоризация-данных-3.5"><span class="toc-item-num">3.5&nbsp;&nbsp;</span>Категоризация данных</a></span></li></ul></li><li><span><a href="#Исследование-различных-групп-заемщиков-на-предмет-надежности" data-toc-modified-id="Исследование-различных-групп-заемщиков-на-предмет-надежности-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Исследование различных групп заемщиков на предмет надежности</a></span><ul class="toc-item"><li><span><a href="#Есть-ли-зависимость-между-количеством-детей-и-возвратом-кредита-в-срок?" data-toc-modified-id="Есть-ли-зависимость-между-количеством-детей-и-возвратом-кредита-в-срок?-4.1"><span class="toc-item-num">4.1&nbsp;&nbsp;</span>Есть ли зависимость между количеством детей и возвратом кредита в срок?</a></span></li><li><span><a href="#Есть-ли-зависимость-между-семейным-положением-и-возвратом-кредита-в-срок?" data-toc-modified-id="Есть-ли-зависимость-между-семейным-положением-и-возвратом-кредита-в-срок?-4.2"><span class="toc-item-num">4.2&nbsp;&nbsp;</span>Есть ли зависимость между семейным положением и возвратом кредита в срок?</a></span></li><li><span><a href="#Есть-ли-зависимость-между-уровнем-дохода-и-возвратом-кредита-в-срок?" data-toc-modified-id="Есть-ли-зависимость-между-уровнем-дохода-и-возвратом-кредита-в-срок?-4.3"><span class="toc-item-num">4.3&nbsp;&nbsp;</span>Есть ли зависимость между уровнем дохода и возвратом кредита в срок?</a></span></li><li><span><a href="#Как-разные-цели-кредита-влияют-на-его-возврат-в-срок?" data-toc-modified-id="Как-разные-цели-кредита-влияют-на-его-возврат-в-срок?-4.4"><span class="toc-item-num">4.4&nbsp;&nbsp;</span>Как разные цели кредита влияют на его возврат в срок?</a></span></li><li><span><a href="#Возможные-причины-появления-большого-количества-пропусков-в-исходных-данных" data-toc-modified-id="Возможные-причины-появления-большого-количества-пропусков-в-исходных-данных-4.5"><span class="toc-item-num">4.5&nbsp;&nbsp;</span>Возможные причины появления большого количества пропусков в исходных данных</a></span></li><li><span><a href="#Почему-заполнение-пропусков-медианным-значением-—-лучшее-решение-для-количественных-переменных?" data-toc-modified-id="Почему-заполнение-пропусков-медианным-значением-—-лучшее-решение-для-количественных-переменных?-4.6"><span class="toc-item-num">4.6&nbsp;&nbsp;</span>Почему заполнение пропусков медианным значением — лучшее решение для количественных переменных?</a></span></li></ul></li><li><span><a href="#Общий-вывод" data-toc-modified-id="Общий-вывод-5"><span class="toc-item-num">5&nbsp;&nbsp;</span>Общий вывод</a></span></li></ul></div>

# Исследование надежности заемщиков


## Описание проекта

  Перед нами статистика о платёжеспособности клиентов. Результаты данного исследования можно использовать при построении модели *кредитного скоринга* — специальной системы, которая оценивает способность потенциального заёмщика вернуть кредит банку.

### Описание данных
**Данный датасет имеет следующие столбцы:**
- `children` — количество детей в семье
- `days_employed` — общий трудовой стаж в днях
- `dob_years` — возраст клиента в годах
- `education` — уровень образования клиента
- `education_id` — идентификатор уровня образования
- `family_status` — семейное положение
- `family_status_id` — идентификатор семейного положения
- `gender` — пол клиента
- `income_type` — тип занятости
- `debt` — имел ли задолженность по возврату кредитов
- `total_income` — ежемесячный доход
- `purpose` — цель получения кредита

### Цель исследования:
В процессе нашей исследовательской работы перед нами стоит три основные задачи:
- Выяснить как влияет количество детей клиента на факт погашения кредита в срок;
- Установить влияет ли пол клиента на факт погашения кредита в срок;
- Установить три самые популярные цели кредитов среди клиентов старше 30 лет

### Ход исследования:
- Обзор данных файла *data.csv*;
- Предобработка данных:
  1. Работа с пропусками;
  2. Обработка аномальных значений;
  3. Изменение типов данных;
  4. Обработка дубликатов;
  5. Категоризация данных
- Исследование различных групп заемщиков на предмет надежности:
  1. Есть ли зависимость между количеством детей и возвратом кредита в срок?;
  2. Есть ли зависимость между семейным положением и возвратом кредита в срок?;
  3. Есть ли зависимость между уровнем дохода и возвратом кредита в срок?;
  4. Как разные цели кредита влияют на его возврат в срок?;
  5. Возможные причины появления большого количества пропусков в исходных данных;
  6. Почему заполнение пропусков медианным значением — лучшее решение для количественных переменных?
- Общий вывод

## Общая информация о данных

Импортируем необходимые библиотеки и прочитаем файл:

In [6]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

Выведем на экран первые 20 строчек датафрейма `data`:

In [3]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

Обзорный просмотр данных датасета показал следующее:
- В столбце `days_employed` встречаются как аномальные значения такие как отрицательный стаж работы, так и пропущенные значения;
- В столбце `education` встречается различное написание одних и тех же значений, что свидетельствует о наличии неявных дубликатов;
- В столбце `total_income` присутствуют пропущенные значения;

Выведем на экран основную информацию о датафрейме с помощью метода info():

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Работа с пропущенными значениями

Выведем количество пропущенных значений для каждого столбца:

In [5]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

В нашем датасете имеются пропуски в столбцах `days_employed`(общий трудовой стаж в днях) и `total_income`(ежемесячный доход):

Начнем работу с пропущенными значениями со столбца `total_income`, содержащим информацию о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому будет справедливым заполнить пропуски  медианным значением по каждому типу из столбца `income_type`(тип занятости):

In [7]:
for inc in data['income_type'].unique():
    data.loc[(data['income_type'] == inc) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == inc), 'total_income'].median()

Заполним пропуски в столбце `days_employed` медианными значениями по каждому типу занятости `income_type`:

In [8]:
for e in data['income_type'].unique():
    data.loc[(data['income_type'] == e) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == e), 'days_employed'].median()

Убедимся, что все пропуски заполнены и перейдем к следующему этапу - обработке аномальных значений:

In [9]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Обработка аномальных значений

В данных столбца `days_employed` встречаются аномальные значения, которые не отражают действительность и появились в результате какой-то ошибки. Обработаем значения в этом столбце, заменив все отрицательные значения положительными с помощью метода abs():

In [10]:
data['days_employed'] = data['days_employed'].abs()

Для каждого типа занятости выведем медианное значение трудового стажа `days_employed` в днях:

In [13]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получились аномально большие значения трудового стажа. Сложно установить в результате чего появились такие аномальные данные. Исправить их не представляется возможным, поэтому оставим их как есть. В любом случае этот столбец не понадобится вам для исследования.

В поисках аномальных значений рассмотрим следующий столбец `children`:

In [14]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

В столбце `children` есть два аномальных значения - `-1` и `20`. Удалим строки, в которых встречаются такие аномальные значения:

In [15]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

### Изменение типов данных

Заменим вещественный тип данных в столбце `total_income` на целочисленный с помощью метода astype():

In [16]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

Обработаем неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному. Приведем все к нижнему регистру. Затем проверим остальные столбцы на наличие дубликатов:

In [35]:
data['education'] = data['education'].str.lower()

Выведем на экран количество строк-дубликатов в данных, а затем удалим их:

In [36]:
data.duplicated().sum()

0

In [37]:
data = data.drop_duplicates()

### Категоризация данных

На основании диапазонов, указанных ниже, создадим в датафрейме `data` столбец `total_income_category` со следующими категориями:
- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.

Для категоризации данных предварительно напишем следующую функцию:

In [38]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

Применим данную функцию и создадим новый столбец `total_income_category`:

In [39]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

Выведем на экран перечень уникальных целей взятия кредита из столбца `purpose`:

In [40]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

Создадим функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

Изучим данные в столбце `purpose` и определим, какие именно подстроки помогут нам правильно определить категорию:

In [41]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

Применим данную функцию и создадим новый столбец `purpose_category`:

In [42]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

## Исследование различных групп заемщиков на предмет надежности

### Есть ли зависимость между количеством детей и возвратом кредита в срок?

Построим сводную таблицу, чтобы увидеть как зависит надежность заемщика от количества детей. Для этого посчитаем количество заемщиков в каждой группе и количество должников, а затем посчитаем конверсию вернувших кредит в срок:

In [46]:
grouped_by_children = data.pivot_table(index='children',values='debt',aggfunc=['count','sum','mean'])
grouped_by_children.columns = ['Количество заемщиков', 'Количество должников', 'Доля невозврата']
grouped_by_children.index = (['Дети отсутствуют', 'Один ребенок', 'Двое детей', 'Трое детей', 'Четверо детей'
                              , 'Пятеро детей'])
print(grouped_by_children)

                  Количество заемщиков  Количество должников  Доля невозврата
Дети отсутствуют                 14091                  1063         0.075438
Один ребенок                      4808                   444         0.092346
Двое детей                        2052                   194         0.094542
Трое детей                         330                    27         0.081818
Четверо детей                       41                     4         0.097561
Пятеро детей                         9                     0         0.000000


**В результате исследования данного вопроса мы выяснили следующее:**
  1. Доля невозврата среди заемщиков, у которых нет детей, составляет 7,54%, что является наименьшим значением среди сравниваемых категорий заемщиков. Тем самым делает эту категорию самой надежной;
  2. Количество должников увеличивается с ростом количества детей. Среди заемщиков, у которых один ребенок, процент должников составляет 9.23%, тогда как у заемщиков, которых двое детей, процент невозврата достигает значения 9.45%;
  3. У заемщиков с тремя детьми доля невозврата составляет 8.18%, что несколько противоречит выявленной раннее зависимости доля невозврата по кредитам от количества детей. Однако данная аномалия могла возникнуть по причине небольшой по размеру выборки. Количество заемщиков составляет всего 330 человек, тогда как количестов замщиков без детей,с одним ребенком и с двумя составляет 14091, 4808 и 2052 человек соответственно. Тем не менее доля невозврата больше процентного соотношения должников среди заемщиков, у которых нет детей, на 0.64%;
  4. У заемщиков с четырьмя детьми доля невозврата также показывает зависимость роста числа должников при увеличении количества детей и составляет - 9.76%. Однако к этим данным стоит относиться с осторожностью, так как величина выборки сравнительно невелика и составляет всего 41 человек;
  5. В группе заемщиков, у которых пять детей, данных недостаточно для анализа. Величина выборки составляет всего 9 человек при доле невозврата в 0%. На основании этих данных можно сделать ошибочный вывод о том, что самыми надежными клиентами банка являются многодетные заемщики с количеством детей равным пяти. Таким образом исключаем эту группу из нашего анализа.

### Есть ли зависимость между семейным положением и возвратом кредита в срок?

Построим сводную таблицу, чтобы увидеть как зависит надежность заемщика от семейного статуса. Для этого посчитаем количество заемщиков в каждой группе и количество должников, а затем посчитаем конверсию вернувших кредит в срок:

In [44]:
grouped_by_family_status = data.pivot_table(index='family_status',values='debt',aggfunc=['count','sum','mean'])
grouped_by_family_status.columns = ['Количество заемщиков', 'Количество должников', 'Доля невозврата']
print(grouped_by_family_status)

                       Количество заемщиков  Количество должников  \
family_status                                                       
Не женат / не замужем                  2796                   273   
в разводе                              1189                    84   
вдовец / вдова                          951                    63   
гражданский брак                       4134                   385   
женат / замужем                       12261                   927   

                       Доля невозврата  
family_status                           
Не женат / не замужем         0.097639  
в разводе                     0.070648  
вдовец / вдова                0.066246  
гражданский брак              0.093130  
женат / замужем               0.075606  


**По результатам исследования данного вопроса было установлено следующее:**
  1. Самыми надежными являются заемщики в семейному статусе "вдовец / вдова". Доля должников составляет 6.62% от общего числа заемщиков, имеющих данный статус;
  2. Самой неплатежеспособной категорией являются клиенты со статусом "Не женат / не замужем". Их доля невозврата составляет 9.76%;
  3. Заемщики, состоящие в гражданском браке, также показали себя как ненадежные заемщики. Доля невозврата среди них - 9.31%;
  4. Доля невозврата среди клиентов с семейными стутусами "в разводе" и "женат / замужем" составляет соответственно 7.06% и 7.56%. Таким образом данные категории клиентов также можно отнести к группе надежных заемщиков.

### Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

Построим сводную таблицу, чтобы проследить как зависит надежность заемщика от его уровня дохода. Для этого посчитаем количество заемщиков в каждой группе и количество должников, а затем посчитаем конверсию вернувших кредит в срок:

In [45]:
grouped_by_total_income_category = (data.pivot_table(index='total_income_category'
                                                     ,values='debt',aggfunc=['count','sum','mean']))
grouped_by_total_income_category.columns = ['Количество заемщиков', 'Количество должников', 'Доля невозврата']
grouped_by_total_income_category.index = ['Доход более 1 млн.руб.', 'Доход от 200 тыс.руб. до 1 млн.руб.',\
                             'Доход от 50 до 200 тыс.руб.', 'Доход от 30 до 50 тыс.руб.', 'Доход до 30 тыс.руб']
print(grouped_by_total_income_category)

                                     Количество заемщиков  \
Доход более 1 млн.руб.                                 25   
Доход от 200 тыс.руб. до 1 млн.руб.                  5014   
Доход от 50 до 200 тыс.руб.                         15921   
Доход от 30 до 50 тыс.руб.                            349   
Доход до 30 тыс.руб                                    22   

                                     Количество должников  Доля невозврата  
Доход более 1 млн.руб.                                  2         0.080000  
Доход от 200 тыс.руб. до 1 млн.руб.                   354         0.070602  
Доход от 50 до 200 тыс.руб.                          1353         0.084982  
Доход от 30 до 50 тыс.руб.                             21         0.060172  
Доход до 30 тыс.руб                                     2         0.090909  


**В результате исследования данного вопроса мы выяснили:**
  1. Самыми надежными заемщиками являются клиенты с уровнем дохода от 30 до 50 тыс.рублей. Доля должников среди них составляет 6.02%;
  2. На втором месте - заемщики с уровнем дохода от 200 тыс.руб. до 1 млн.рублей. Доля невозврата среди этой категории клиентов составляет 7.06%;
  3. Третьими по очереди надежных заемщиков идут клиенты с доходом от 1 млн.рублей. Из 25 человек только двое не вернули кредит в срок. Доля должников составляет 8.00% от общего числа клиентов данной категории;
  4. Самыми ненадежными оказались клиенты с уровнем дохода от 50 до 200 тыс.руб и с доходом от 30 тыс.рублей. Их доля невозврата составила 8.50% и 9.09% соответсвенно.

Расмотрим подробнее группу заемщиков с доходами менее 30 тыс.рублей:

Отберем из массива данных должников с доходом менее 30 тыс.рублей. Создадим сводную в таблицу, в которой в качестве индекса будет выступать социальный статус заемщика,а в столбцах занесены соответственно значения общего количества заемщиков, количества должников и доли невозврата,разбитые по группам:

In [47]:
grouped_by_total_under_30 = data.loc[(data['total_income_category']=='E') & (data['debt']==1)]
grouped_by_total_under_30 = grouped_by_total_under_30[['income_type','total_income','purpose_category']]
grouped_by_total_under_30.columns = ['Социальный статус', 'Ежемесячный доход', 'Цель кредита']
print(grouped_by_total_under_30.reset_index(drop=True))

  Социальный статус  Ежемесячный доход              Цель кредита
0         пенсионер              29154  операции с недвижимостью
1         пенсионер              20667  операции с недвижимостью


Оба должника оказались пенсионерами c ежемесячные доходами 29154 руб. и 20667 руб. соответственно. Цель кредита - операции с недвижимостью. Выдавать кредит на дорогостоящие операции с недвижимостью было крайне рискованным решением, учитывая уровень дохода заемщиков.

### Как разные цели кредита влияют на его возврат в срок?

Построим сводную таблицу, чтобы проследить как зависит надежность заемщика от цели кредита. Для этого посчитаем количество заемщиков в каждой группе и количество должников, а затем посчитаем конверсию вернувших кредит в срок:

In [48]:
grouped_by_purpose_category = data.pivot_table(index='purpose_category',values='debt',aggfunc=['count','sum','mean'])
grouped_by_purpose_category.columns = ['Количество заемщиков', 'Количество должников', 'Доля невозврата']
print(grouped_by_purpose_category)

                          Количество заемщиков  Количество должников  \
purpose_category                                                       
операции с автомобилем                    4279                   400   
операции с недвижимостью                 10751                   780   
получение образования                     3988                   369   
проведение свадьбы                        2313                   183   

                          Доля невозврата  
purpose_category                           
операции с автомобилем           0.093480  
операции с недвижимостью         0.072551  
получение образования            0.092528  
проведение свадьбы               0.079118  


**В результате изучения данного вопроса мы выяснили:**
  1. Клиенты берущие кредит на операции с недвижимостью чаще всего выплачивают кредит в срок. Лишь 7.26% заемщики не вернули кредит в срок;
  2. Заемщики берущие кредит на проведение свадьбы также являются надежными клиентами банка. Процент невозврата по таким кредитам составляет 7.91%;
  3. Самыми ненадежными клиентами оказались те, которые взяли кредиты на получение образования и операции с автомобилем. Процент невозврата по таким кредитам составил 9.25% и 9.35%, что делает выдачу кредитов на эти цели не такой привлекательной для банка.  

### Возможные причины появления большого количества пропусков в исходных данных

**В контексте нашего датасета могут быть следующие причины появления пропущенных значения:**
  1. Среди категории "Безработные" вероятнее всего есть те, кто ни одного дня не работал. Поэтому графа стаж `days_employed` не была заполнена;
  2. Среди категории "Пенсионеры" вероятно есть те, кто по ошибке не указал уровень дохода, считая пенсию социальной льготой,а не личным доходом;
  3. Есть вероятность, что для категории "Пенсионеры" было сложно посчитать свой стаж в днях. Необходимо редактировать форму таким образом, чтобы вводя стаж в годах - в таблицу заносились данные в днях;
  4. Отсутствие желания или возможностей у заемщиков документально подтвердить информацию о доходе и стаже.

### Почему заполнение пропусков медианным значением — лучшее решение для количественных переменных?

    Медиана корректно характеризует данные, когда некоторые значения сильно выделяются среди большинства. Медиана устойчива к выбросам данных. 

    Среднее арифметическое значение более устойчиво к неоднородной по значениям выборке, когда большая часть элементов расбросаны по числовому лучу.

## Общий вывод

**В результате исследования мы можем сделать следующие выводы:**
  1. Самыми надежными заемщиками являются клиенты не имеющие детей. С увеличением количества детей возрастает вероятность невозврата кредита в срок;
  2. Рассматривая группы разбитые по семейному статусу, можно прийти к выводу о том, что самыми надежными плательщиками являются клиенты категории "вдовец / вдова";
  3. Менее надежными, но тем не менее с относительно небольшим процентом невозврата в срок являются категории "В разводе" и "Женат / Замужем";
  4. Чаще остальных просрочили выплату по кредиту заемщики состоящие в гражданском браке и клиенты категории "Не женат / не замужем";
  5. Если рассматривать заемщиков, разбивая их на группы по ежемесячному доходу,то можно прийти к выводу, что самыми ненадежными клиентами являются те, чей доход не превышает 30 тыс.рублей, а также те, чей доход составляет от 50 до 200 тыс.руб;
  6. Наименьший процент невыплаты в срок имеют клиенты с уровнем дохода от 30 до 50 тыс.руб;
  7. На втором и третьем местах по надежности идут заемщики с доходом от 200 тыс.руб. до 1 млн.руб и с доходом более 1 млн.рублей;
  8. Клиенты берущие кредит на операции с недвижимостью и проведение свадьбы чаще всего выплачивают кредит в срок;
  9. Самыми ненадежными заемщиками оказались те, которые взяли кредиты на получение образования и операции с автомобилем. 